server/controllers/Profile.js

```diff
@@ -86,6 +87,64 @@ exports.getPublicProfile = asyncHandler(async (req, res, next) => {
   res.send({ profile });
 });
 
+// @route GET /profiles
+// @desc List of all profiles
+// @access Public
+exports.getProfiles = asyncHandler(async (req, res) => {
+  const profiles = await Profile.find(
+    { isSitter: true },
+    "_id userId firstName lastName photoUrl description address"
+  );
+  res.send({ profiles });
+});
+
+// @route GET /profile
+// @desc Returns full profile for auth user
+// @access Public
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  if (userId) {
+    if (!mongoose.Types.ObjectId.isValid(userId)) {
+      return res.status(400).send("Bad Request");
+    }
+  }
+
+  const resp = await Profile.findOne({ userId });
+  if (!resp) {
+    res.status(404);
+    throw new Error("Invalid profile id");
+  }
+
+  res.send({
+    success: {
+      _id: resp._id,
+      userId: resp.userId,
+      firstName: resp.firstName,
+      lastName: resp.lastName,
+      gender: resp.gender,
+      birthDate: resp.birthDate,
+      email: resp.email,
+      phoneNum: resp.phoneNum,
+      address: resp.address,
+      description: resp.description,
+    },
+  });
+});
+
+// @route GET /public-profile
+// @desc Returns public user profile
+// @access Public
+exports.getPublicProfile = asyncHandler(async (req, res, next) => {
+  const userId = req.params.id;
+
+  const profile = await Profile.findOne(
+    { userId },
+    "_id userId firstName lastName photoUrl description address"
+  );
+  res.send({ profile });
+});
+
```

Comment:
> Just noticed that these are duplicated within the file for some reason 🤔

Must because of rebasing error. Gona fix it.

Reply:
> Just noticed that these are duplicated within the file for some reason 🤔

No it's not duplicated. getUprofile is a private method to return current user profile. getprofiles and getallprofiles are from Robert, he use these function to get all user profiles a certain profile with userid and .et all user profiles
getUprofile cannot be replaced with getprofiles as that time because of datastructure, the _id property of user cannot be got